### DISCUSSION TITLE
Installing IPAs over-the-air

### DISCUSSION THREAD

**Discussion by MurrayHamilton** [airsdk/Adobe-Runtime-Support#3449] [General]
The lightning port on my iPhone 13 has broken so I can't connect it to my PC any more. Is there any way to install IPAs over-the-air? Can AIR do it itself? I've got access to a Mac if that helps. Thanks!

P.S. I've cleared out the fluff from the port and tried about 5 different cables (that all used to work). It used to work at a certain angle but that's not the case anymore 😢

**Comment by itlancer** [comment]
You can try this service https://www.diawi.com/

**Reply by MurrayHamilton** [reply]
Thanks for the link, but I'd rather not upload the IPA to a 3rd party service. Something that works locally would be preferable (sorry, should have said that in my original post).

**Comment by ajwfrost** [comment]
If you put the mac and the phone on the same wifi network, then you should be able to access the phone from Xcode -> Window menu -> Devices and Simulators.
The "+" button in the "Installed Apps" can be used to browse for an IPA file..

**Reply by MurrayHamilton** [reply]
Thanks for this - I had to connect my phone to my Mac via USB first to enable the option to do this (the Connect via network tick box). Fortunately I managed to jam my lightning cable into my phone at the right angle and with enough force (i.e. LOADS!) to get it to connect - don't know how many more times I'll get away with that though!

Anyway, I tried to install a dev build IPA with the + button but got this error:
```
Error installing  '/Volumes/XXXXXX.ipa', ERROR: Error Domain=com.apple.dt.CoreDeviceError Code=3002 "Failed to install the app on the device." UserInfo={NSUnderlyingError=0x6000017494a0 {Error Domain=com.apple.dt.CoreDeviceError Code=3000 "The item at XXXXXX.ipa is not a valid bundle." UserInfo={NSURL=file:///Volumes/XXXXXX.ipa, NSLocalizedFailureReason=Failed to read the bundle., NSLocalizedDescription=The item at XXXXXX.ipa is not a valid bundle.}}, NSLocalizedDescription=Failed to install the app on the device., NSURL=file:///Volumes/XXXXXX.ipa}
Domain: com.apple.dt.DVTCoreDevice
Code: -1
User Info: {
    DVTErrorCreationDateKey = "2024-09-09 12:53:26 +0000";
}
--


System Information

macOS Version 14.4.1 (Build 23E224)
Xcode 15.0.1 (22266) (Build 15A507)
Timestamp: 2024-09-09T13:53:26+01:00
```

This build has been fine to install via Windows when my port used to work.

**Reply by ajwfrost** [reply]
I'm wondering whether there's an issue with macOS security - given the macOS version you have there... could you try putting the IPA file into your "Documents" or "Downloads" folder and then trying from there?

thanks

**Reply by MurrayHamilton** [reply]
Tried both of those folders, no luck I'm afraid!

**Reply by ajwfrost** [reply]
Hmmm... sounds like it really may not be a valid bundle format .. are you able to open it within a zip program (or on the terminal command-line with something like "unzip -l file.ipa") to check the contents? Should be a folder "Payload" within which is your "appname.app" folder, and in there an "Info.plist" file plus your executable, plus lots of other files!

**Reply by MurrayHamilton** [reply]
Yeah I can unzip via terminal and get everything you described

**Reply by MurrayHamilton** [reply]
So I manged to get this working by using this method:

1. Changing the ".ipa" file extension to ".zip"
2. Open/unzip
3. Install the .app file located in resulting "Payload" folder instead

From here: https://forums.developer.apple.com/forums/thread/738233?answerId=767215022#767215022

**Comment by jeuxdaily** [comment]
what about Bluetooth??

**Reply by MurrayHamilton** [reply]
Thanks - I've sent an IPA over AirDrop from my Mac, both a dev and ad-hoc build, but I can't install it on my phone (it's just sitting in Files with no option to do anything). So I'd assume sending it via Bluetooth would give a similar result...?

I read around a bit on AirDropping it, and it might be that you need an Enterprise developer account (which I don't have) to be able to install IPAs in this way.
